src/org/mozilla/javascript/NativeObject.java

```diff
@@ -567,6 +567,13 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
               Scriptable sourceObj = ScriptRuntime.toObject(cx, thisObj, args[i]);
               Object[] ids = sourceObj.getIds();
               for (Object key : ids) {
+                if (targetObj instanceof ScriptableObject) {
+                  ScriptableObject desc = ((ScriptableObject) targetObj).getOwnPropertyDescriptor(cx, key);
+                  if (desc != null && isDataDescriptor(desc) &&
+                      Boolean.FALSE.equals(desc.get("writable"))) {
```

Comment:
Can you take a look at the other implementation that checks for something like this, which is in ScriptableObject.checkPropertyChange and make it as much like that as you can?
That code uses an "isFalse" method to test for false, which is likely a more consistent way to do this -- these things tend to be very picky for edge cases.

Reply:
Thanks.  I changed to use isFalse. 959147a